eoc_journal/eoc_journal.py

```diff
@@ -97,6 +106,49 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_user_pb_answers_by_section(self):
+        """
+        Returns a list of dicts with pb-answers grouped by section.
+        """
+        # Get the selected blocks and their answers
+        blocks = self.list_pb_answers()
+        course_id = self._get_course_id()
+        user_id = self.runtime.anonymous_student_id
+        answers_names = [block['name'] for block in blocks]
+
+        answers = Answer.objects.filter(  # pylint: disable=no-member
+            course_key=course_id,
+            student_id=user_id,
+            name__in=answers_names
+        )
+
+        if answers.count() > 0:
+            # Map answer names to student inputs
+            students_inputs = {a.name: a.student_input for a in answers if a.student_input}
+
+            # Group answers by section
+            answers = OrderedDict()
+
+            for block in blocks:
+                name = block['name']
+                section = block['section']
+
+                if section not in answers:
+                    answers[section] = []
+
+                answers[section].append({
+                    'answer': students_inputs[name],
```

Comment:
> I just replaced the '/' with _("Not answered yet.").

Thanks, that's better!

> Unanswered questions won't. show in Apros. / lms

Won't they still show up if user goes straight to the EOC journal the first time they open the course (skipping sections with problem builder answers)? It's an edge case, but I think it's possible, especially while testing.